src/Microsoft.AspNetCore.StaticFiles/StaticFileContext.cs

```diff
@@ -425,5 +433,62 @@ private ContentRangeHeaderValue ComputeContentRange(RangeItemHeaderValue range,
             length = end - start + 1;
             return new ContentRangeHeaderValue(start, end, _length);
         }
+
+        //internal for testing
+        internal static void ApplyCacheProfile(HttpContext context, CacheProfile cacheProfile)
+        {
+            if (context == null)
+            {
+                throw new ArgumentNullException(nameof(context));
+            }
+            if (cacheProfile == null)
+            {
+                throw new ArgumentNullException(nameof(cacheProfile));
+            }
+
+            var headers = context.Response.Headers;
+
+            if (!string.IsNullOrEmpty(cacheProfile.VaryByHeader))
+            {
+                headers[HeaderNames.Vary] = cacheProfile.VaryByHeader;
+            }
+
+            if (cacheProfile.NoStore)
+            {
+                headers[HeaderNames.CacheControl] = "no-store";
+
+                // Cache-control: no-store, no-cache is valid.
+                if (cacheProfile.Location == ResponseCacheLocation.None)
+                {
+                    headers.AppendCommaSeparatedValues(HeaderNames.CacheControl, "no-cache");
+                    headers[HeaderNames.Pragma] = "no-cache";
+                }
+            }
+            else
+            {
+                string cacheControlValue = null;
```

Comment:
for a null? It also looks kind of weird if you cast i.e. `var cacheControlValue = (string)null;`.